tests/unit/db/database_tests.py

```diff
@@ -299,19 +299,28 @@ def test_retrieve_raw_view_results(self):
         self.assertIsInstance(raw_rslt, dict)
         self.assertEqual(len(raw_rslt.get('rows')), 100)
 
-    def test_all_docs(self):
+    def test_all_docs_post(self):
         """
-        Test the all_docs functionality
+        Test the all_docs POST request functionality
+        """
+        keys_list = list(map(lambda x: x['id'], self.populate_db_with_documents()))
```

Comment:
IIUC this keys list is effectively all 100 test documents. I think this means the assertions that this test makes would work even if the keys argument was completely ignored.
It might be worth having an additional test (or changing the list of keys in this test) to use a sub-set to validate that the keys argument it is actually being used.